toolium/utils/dataset.py

```diff
@@ -594,22 +596,42 @@ def get_value_from_context(param, context):
     """
     parts = param.split('.')
     value = _get_initial_value_from_context(parts[0], context)
+    msg = None
 
     for part in parts[1:]:
         if isinstance(value, dict) and part in value:
             value = value[part]
+        elif isinstance(value, list) and part.isdigit() and int(part) < len(value):
+            value = value[int(part)]
         elif hasattr(value, part):
             value = getattr(value, part)
         else:
-            if isinstance(value, dict):
-                msg = f"'{part}' key not found in {value} value in context"
-            else:
-                msg = f"'{part}' attribute not found in {type(value).__name__} class in context"
+            msg = get_value_context_errors(value, part)
+        if msg:
             logger.error(msg)
             raise Exception(msg)
     return value
 
 
+def get_value_context_errors(value, part):
```

Comment:
I would rather make it private, it does not seem very reusable outside this module.
```suggestion
def _get_value_context_error_msg(value, part):
```